### ISSUE TITLE
Slicing: cannot slice along batch axis

### ISSUE THREAD

**Issue by calclavia** [onnx/onnx-coreml#379]
When trying to do a simple slice in the model:

e.g:
```
return x[:, :, :-self.__padding]
```

I get this error. Somehow the converter thinks that a slice is occurring in the batch dimension when in reality it's not in the batch dimension.
```
2/6: Converting Node Type Slice
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/henry/code/altum/model-trainer/export/export.py", line 34, in <module>
    mlmodel = convert(onnx_model)
  File "/usr/local/lib/python3.6/site-packages/onnx_coreml/converter.py", line 458, in convert
    _convert_node(builder, node, graph, err)
  File "/usr/local/lib/python3.6/site-packages/onnx_coreml/_operators.py", line 1755, in _convert_node
    return converter_fn(builder, node, graph, err)
  File "/usr/local/lib/python3.6/site-packages/onnx_coreml/_operators.py", line 1288, in _convert_slice
    err.unsupported_op_configuration(builder, node, graph, 'cannot slice along batch axis')
  File "/usr/local/lib/python3.6/site-packages/onnx_coreml/_error_utils.py", line 56, in unsupported_op_configuration
    "Error while converting op of type: {}. Error message: {}\n".format(node.op_type, err_message, )
TypeError: Error while converting op of type: Slice. Error message: cannot slice along batch axis
```

**Comment by aseemw** [comment]
This mode is not supported in CoreML (exactly what is says: "CoreML does not support slice along the batch axis" ), hence the error raised by the converter. 

Do you have a python snippet to reproduce the error, there might be a workaround for the error. 
What is the rank of the model input? 
You maybe be able to use "onnx_coreml_input_shape_map" argument to force a different mapping of the input to CoreML "Seq, batch, channel, height, width" axis and see if that helps.

**Comment by calclavia** [comment]
I'm trying to get a causal convolution working. Here's a module in Pytorch that will trigger the error:
```
class CausalConv1d(torch.nn.Conv1d):
    def __init__(self,
                 in_channels,
                 out_channels,
                 kernel_size,
                 stride=1,
                 dilation=1,
                 groups=1,
                 bias=True):

        super(CausalConv1d, self).__init__(
            in_channels,
            out_channels,
            kernel_size=kernel_size,
            stride=stride,
            padding=0,
            dilation=dilation,
            groups=groups,
            bias=bias)
        
        self.__padding = (kernel_size - 1) * dilation
        
    def forward(self, input):
        result = super(CausalConv1d, self).forward(input) 
        if self.__padding != 0:
            return result[:, :, :-self.__padding]
        return result
```

**Comment by aseemw** [comment]
Can you try using onnx-coreml 0.4.0?
Some of these issues should be fixed in PR #394
